File: src/decoders/news-question-choice.ts
import type { NewsQuestionChoice } from "~/models/news-question-choice";

export const decodeNewsQuestionChoice = (choice: any): NewsQuestionChoice => {
  return {
    value: choice.L,
    position: choice.rang,
    isTextInput: !!choice.estReponseLibre
  };
};

File: library/src/commonMain/kotlin/decoders/news-question-choice.kt
package decoders

import kotlinx.serialization.json.*
import models.NewsQuestionChoice

fun decodeNewsQuestionChoice (choice: JsonObject): NewsQuestionChoice {
    return NewsQuestionChoice(
        value = choice["L"]!!.jsonPrimitive.content,
        position = choice["rang"]!!.jsonPrimitive.int,
        isTextInput = choice["estReponseLibre"]?.jsonPrimitive?.boolean ?: false
    )
}